### DISCUSSION TITLE
Sensor support installation

### DISCUSSION THREAD

**Discussion by WirthmU** [aaronwmorris/indi-allsky#1542] [Q&A]
I wanted to install sensors. In the config section it is stated that "Hardware Device support requires the python GPIO modules". How would I do this? I could not find any help to this.

**Comment by aaronwmorris** [comment]
When you run setup.sh, it will prompt to optionally install the GPIO modules.  If you have already done that, you should be good to go.

**Reply by WirthmU** [reply]
This did the trick - thank you.

I am using the following HAT:

https://www.waveshare.com/wiki/Environment_Sensor_HAT

The BME280 is working already with indi-allsky.

May I ask for the other sensors to be added, too?

**Reply by aaronwmorris** [reply]
The TSL2591 light sensor should work out of the box too.  I will investigate adding the LTR390 UV sensor.

**Reply by WirthmU** [reply]
I can confirm that the TSL2591 is (for MQTT partially - see the curly brackets for sensors 16 and 18 in the picture below) working.
This is MQTT:
![5AD9FB29-EF75-4722-91FF-707AB0F028BF](https://github.com/user-attachments/assets/086ea863-925a-47a7-8ea2-d4267c7a93af)

Could you tell me why there are two places in the config section where MQTT can be configured (MQTT and Sensors)?
I am currently - at least I think - using the definition of the one in the Sensors section.

**Reply by aaronwmorris** [reply]
The MQTT tab is for publishing data to a broker.  The MQTT settings under the Sensors tab is if you want to monitor a sensor that is published to MQTT from another system.

**Reply by WirthmU** [reply]
This is the output of the sensor_test.py script:
```
(indi-allsky) as3@raspberrypi:/home/indi-allsky $ ./misc/sensor_test.py
[WARNING]: Initializing [Sensor A] BME280 I2C temperature device @ 0x76
[WARNING]: Initializing [Sensor B] TSL2591 I2C light sensor device @ 0x29
[WARNING]: No sensor C - Initializing sensor simulator
[INFO]: [Sensor A] BME280 - temp: 36.7c, humidity: 22.4%, pressure: 955.8hPa
[INFO]: Sensor 0: {'dew_point': 11.746054413650835, 'frost_point': 6.320583130909881, 'heat_index': 35.49022191153521, 'data': (36.7248046875, 22.359197504928733, 955.7504542383617)}
[INFO]: [Sensor B] Switching TSL2591 to day mode - Gain 0, Integration: 0
[INFO]: [Sensor B] TSL2591 - lux: 317.9136, visible: 1441884, ir: 22, full: 1441906
[INFO]: Sensor 1: {'sqm_mag': 6.3277866213162755, 'data': (317.91360000000003, 1441884, 22, 1441906)}
[INFO]: Sensor 2: {'data': ()}
```

**Reply by aaronwmorris** [reply]
That seems to be working perfectly!

**Reply by WirthmU** [reply]
I agree, but what the MQTT host receives is not complete:
![BFF20791-C0C3-4E85-9ADF-769F94D04AE8](https://github.com/user-attachments/assets/1f599156-4bd9-44f9-90c1-8405020b3357)

Sensor 16 should be "visible" and sensor 18 should be "full spectrum".

**Reply by aaronwmorris** [reply]
Looks like the `visible` data from TLS2591 is not making it.

**Reply by aaronwmorris** [reply]
I do not understand why it is coming through as the curly braces.

**Reply by aaronwmorris** [reply]
Testing with my TSL2591, I am receiving data properly.

```
$ mosquitto_sub -h localhost -p 1883 -u indi-allsky -P password -t indi-allsky/sensor_user_15 -t indi-allsky/sensor_user_16 -t indi-allsky/sensor_user_17 -t indi-allsky/sensor_user_18
1.7
327683.0
5.0
327688.0
```

**Comment by aaronwmorris** [comment]
Merged #1545 with LTR390 UV light sensor support.  You will need to re-run setup.sh to install the new GPIO module for the sensor.

I do not have this sensor, so I am not able to test the code.  Let me know if it works and I will mark it as "working".

**Reply by WirthmU** [reply]
It works and I get values.

After I did re-boot the host, the MQTT problem is solved, too.

Thank you for your support!
